pkg/reconciler/taskrun/taskrun.go

```diff
@@ -107,6 +115,12 @@ func (c *Reconciler) ReconcileKind(ctx context.Context, tr *v1beta1.TaskRun) pkg
 		// on the event to perform user facing initialisations, such has reset a CI check status
 		afterCondition := tr.Status.GetCondition(apis.ConditionSucceeded)
 		events.Emit(recorder, nil, afterCondition, tr)
+		if sendCloudEvents {
```

Comment:
Any reason for not putting this into `events.Emit` ? That way we wouldn't have to track all `events.Emit` to add that code ?

Reply:
Yes, that was my original plan, but it would require changing the existing signature of Emit events, and I didn't want to make this PR larger than it was already. 
The way I would do it is to pass the context (instead of recorder)  `Emit` (or perhaps add a new method `EmitAll` that takes the `ctx`). The from the context extract the recorder for k8s events and the config for the cloud events.
I can do it in this PR or in a PR on top, wdyt?

Reply:
I would rather do it in this PR as it won't add too much lines of code (aka ain't gonna add 1k :stuck_out_tongue:).

Reply:
Ok, changing that implies that events will also be sent for pipeline runs. 
It also requires some changes to makes the method signature compatible.
I've done that, hopefully it looks good. When a sink is set, it will now start sending cloud events to that sink for all pipeline run and task run events!